pkg/database/database.go

```diff
@@ -87,12 +141,29 @@ func (db *Database) QueryNamespacedResourceByName(ctx context.Context, kind, gro
 	}
 }
 
-func (db *Database) QueryNamespacedCoreResources(ctx context.Context, kind, version, namespace string) ([]*unstructured.Unstructured, error) {
-	return db.performResourceQuery(ctx, db.info.GetNamespacedResourcesSQL(), kind, version, namespace)
+func (db *Database) QueryNamespacedCoreResources(ctx context.Context, kind, version, namespace, limit, continueUUID, continueDate string) ([]*unstructured.Unstructured, string, string, error) {
+	args := []string{kind, version, namespace}
+	query := db.info.GetNamespacedResourcesSQL()
+
+	if limit != "" {
+		if continueUUID == "" || continueDate == "" {
+			query = db.info.GetNamespacedResourcesLimitedSQL()
+			args = append(args, limit)
+		} else {
+			query = db.info.GetNamespacedResourcesLimitedContinueSQL()
+			args = append(args, continueDate, continueUUID, limit)
+		}
+	}
+
```

Comment:
this block is repeated 4 times, could you consider extracting it to a private function?

Reply:
The block is repeated twice: one for the pair `QueryResources` and `QueryCoreResources`, and a second time for the pair `QueryNamespacedResources` and `QueryNamespacedCoreResources`.

So I think the repetition is on the functions instead. Instead of extracting this block out, I prefer  we address the duplication of the whole function (`QueryResources` and `QueryCoreResources` are the same except for `apiVersion := fmt.Sprintf("%s/%s", group, version)`, and the other pair its the same) in another issue.

Reply:
In other words, `QueryResources` is just `QueryCoreResources` with `version = apiVersion`. The same for `QueryNamespacedResources` and `QueryNamespacedCoreResources`, etc.

Reply:
Good point. I agree that the change we should do (at least first) is to refactor the methods so we "merge" `QueryResources` with `QueryCoreResources` and `QueryNamespacedResources` with `QueryNamespacedCoreResources`.

We can create a separate issue that